client/src/main/java/tds/assessment/Item.java

```diff
@@ -198,4 +210,33 @@ public void setPrintable(boolean printable) {
     public void setFormKeys(Set<String> formKeys) {
         this.formKeys = formKeys;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Item item = (Item) o;
+
+        if (fieldTest != item.fieldTest) return false;
+        if (!id.equals(item.id)) return false;
+        if (!itemType.equals(item.itemType)) return false;
+        if (!groupId.equals(item.groupId)) return false;
+        if (!groupKey.equals(item.groupKey)) return false;
+        if (!blockId.equals(item.blockId)) return false;
+        if (!strand.equals(item.strand)) return false;
+        if (itemProperties != null ? !itemProperties.equals(item.itemProperties) : item.itemProperties != null)
+            return false;
+        return formKeys != null ? formKeys.equals(item.formKeys) : item.formKeys == null;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Just wanted to make sure you intended to make it possible for two items to not be equal but produce the same hashcode.